### ISSUE TITLE
[Feature]: New features, ideas and contributions

### ISSUE THREAD

**Issue by Jesulayomy** [Jesulayomy/SCHub#8]
Any new extensions to the project can be discussed, i'll list a few examples of ideas I've been thinking about and would like to implement, it includes expanding the project from the MVP and most of them will create other things to work on. 

Create an issue with the title "[Feature]: <WHat you'd like to implement>" 
If you are creating a feature, also make sure to include any side effects (tasks that could arise from implementing it). Ex: Creating a new api route for books would affect models, api, data, docs. If you did not work on either of those in your PR, then list them or create the issues, mention the issue you created that caused this side effect in your PR.

Ideas

- [ ] Student page course teacher popup showing teacher contact
- [ ] Adding schedules / time table with all the courses and times, which can be displayed on dashboards for teachers and students (This will be pretty large)
- [ ] Metrics (with functions or something similar)
- [ ] Feature
- [ ] Feature
- [ ] Feature
- [ ] Feature
- [ ] Feature
- [ ] Major feature
- [ ] major feature
- [ ] Major feature
- [ ] Suggested feature
- [ ] Suggested feature
- [ ] Suggested feature
- [ ] Suggested feature
- [ ] Minor feature
- [ ] Minor feature
- [ ] Minor feature

**Comment by LoayTarek5** [comment]
Hi @Jesulayomy, i hope you are doing well
"Could you please assign me this issue? I will implement any features."

**Comment by Jesulayomy** [comment]
@LoayTarek5 Do you want to implement a specific feature?
You would have to create an issue for your specific feature.

**Comment by LoayTarek5** [comment]
@Jesulayomy  i think the first one:
- [ ] Student page course teacher popup showing teacher contact

**Comment by LoayTarek5** [comment]
@Jesulayomy  
when i try the website i cannot register as a new student

**Comment by Jesulayomy** [comment]
Okay @LoayTarek5, you'd need to use Axios, React state, HTML, CSS for that. You would call the teacher API using the result from courses fetched using axios.
Check the `schub/src/pages/StudentDashboard.js` file, you would work there along with css I guess.

Once you get the gist of it create your issue with some process details and I will assign it to you.

Yes you cannot register yourself (admins can do that), there are login details in my README.md#Public Access, so you can see how it looks now.
```python
{
  "admin": {
    "email": "tester@schub.com",
    "password": "testerpwd"
  },
  "teacher": {
    "email": "TUANN@schub.com",
    "password": "TUANNPWD"
  },
  "student": {
    "email": "GREAMEM@schub.com",
    "password": "GREAMEMPWD"
  }
}
```

**Comment by LoayTarek5** [comment]
@Jesulayomy to Full understand 
Do you mean the final result you want is for the "show teacher" button to display the teacher's contact when pressed?
![image](https://github.com/user-attachments/assets/985519d1-dae3-4eec-b7f4-b9e0900919fd)

**Comment by Jesulayomy** [comment]
Yes, currently it just shows the name under, I think a modal popup would look more presentable yea? If you think something else let me know. also you can reach me on discord: https://discord.com/users/488238079753060353

**Comment by LoayTarek5** [comment]
@Jesulayomy Okay, I will work on that.🐱‍🏍

**Comment by Jesulayomy** [comment]
Thanks! Will expect to see your great work soon 🥇

**Comment by LoayTarek5** [comment]
@Jesulayomy i have some problems when run the commands in the READme 
first: pre-commit install
it gives me"bash: pre-commit: command not found"
second cat data/setup_dev_db.sql | mysql -u root -p gives me "Enter password: *****
ERROR 1045 (28000): Access denied for user 'root'@'localhost' (using password: YES)"

**Comment by Jesulayomy** [comment]
Hi @LoayTarek5 Sorry for the mixup, I must have skipped pushing the update with pre-commit in the requirements.txt file. I will fix that right away. You can temporarily fix with `pip install pre-commit`. 

The mysql command depends on how you have mysql setup on your device. You may be able to use `sudo mysql` in places where `mysql -u root -p` is used instead.

**Comment by LoayTarek5** [comment]
@Jesulayomy i am sorry, but I am having some troubles with MySQL at the moment. Therefore, you can see another one to solve this issue 
I apologize for the inconvenience.
